MovieApp/MovieCategoriesViewController.swift

```diff
@@ -0,0 +1,266 @@
+        trendingLabel.autoPinEdge(.top, to: .bottom, of: freeCollectionView, withOffset: 25)
+        trendingLabel.attributedText = NSMutableAttributedString().bold("Trending", fontSize: 20)
+        
+        trendingCollectionView.autoPinEdge(.leading, to: .leading, of: whatsPopularLabel, withOffset: 0)
+        trendingCollectionView.autoPinEdge(.top, to: .bottom, of: trendingLabel, withOffset: -15)
+        trendingCollectionView.autoSetDimension(.height, toSize: 220)
+        trendingCollectionView.autoPinEdge(.trailing, to: .trailing, of: view)
+        trendingCollectionView.delegate = self
+        trendingCollectionView.dataSource = trendingCollectionViewDataSource
+        
+    }
+    
+    private func registerCollectionViews() {
+
+        for popularMovie in movieDetails.popularMovies{
+            popularCollectionView.register(MovieCategoriesViewCell.self, forCellWithReuseIdentifier: popularMovie.name)
+        }
+ 
+        
+        for freeMovie in movieDetails.freeToWatchMovies{
+            freeCollectionView.register(MovieCategoriesViewCell.self, forCellWithReuseIdentifier: freeMovie.name)
+        }
+        
+        for trendingMovie in movieDetails.trendingMovies{
+            trendingCollectionView.register(MovieCategoriesViewCell.self, forCellWithReuseIdentifier: trendingMovie.name)
+        }
+        
+    }
+    
+    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+        return .portrait
+    }
+
+    override var shouldAutorotate: Bool {
+        return true
+    }
+    
+}
+
+
+class MovieCategoriesViewCell: UICollectionViewCell {
+    
+    let thumbnailImageView = UIImageView()
+    let favoriteButton = UIButton()
+    
+    override init (frame: CGRect) {
+        super.init(frame: frame)
+
+        
+        contentView.addSubview(thumbnailImageView)
+        contentView.addSubview(favoriteButton)
+        contentView.backgroundColor = .red
+        contentView.layer.cornerRadius = 15
+        contentView.layer.masksToBounds = true
+    
+        thumbnailImageView.autoPinEdgesToSuperviewEdges()
+        thumbnailImageView.contentMode = .scaleAspectFill
+        thumbnailImageView.clipsToBounds = true
+        
+        favoriteButton.autoPinEdge(.leading, to: .leading, of: contentView, withOffset: 10)
+        favoriteButton.autoPinEdge(.top, to: .top, of: contentView, withOffset: 10)
+        favoriteButton.autoSetDimension(.width, toSize: 30)
+        favoriteButton.autoSetDimension(.height, toSize: 30)
+        favoriteButton.backgroundColor = .darkGray
+        favoriteButton.layer.cornerRadius = 15
+        favoriteButton.layer.opacity = 0.8
+        let homeSymbolConfiguration = UIImage.SymbolConfiguration(pointSize: 17, weight: .regular)
+        let homeImage = UIImage(systemName: "heart", withConfiguration: homeSymbolConfiguration)
+        favoriteButton.setImage(homeImage, for: .normal)
+
+        contentView.layer.cornerRadius = 12
```

Comment:
All this should be extracted to separate methods. Cells are views too. So you can use `createViews`, `styleViews` and `defineLayout` methods here.